torch/csrc/api/include/torch/nn/rnn.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <torch/nn/module.h>
+#include <torch/nn/rnn.h>
+#include <torch/tensor.h>
+
+#include <ATen/ATen.h>
+
+#include <vector>
+
+namespace torch { namespace nn {
+
+// This is largely just a proof-of-concept of the interface.
+class LSTM : public torch::nn::CloneableModule<LSTM> {
+ public:
+  LSTM(long input_features, long state_size)
+      : CloneableModule<LSTM>("LSTM"),
```

Comment:
The fact that you need to do
```cpp
MyModule() : ClonableModule<MyModule>("MyModule")
```
is a bit annoying. The density of `MyModule` is really high.

Is `clone` really that important? The non-CRTPed version would look so much nicer